qiskit/algorithms/eigensolvers/numpy_eigensolver.py

```diff
@@ -257,36 +237,35 @@ def compute_eigenvalues(
             eigvals = []
             aux_ops = []
             cnt = 0
-            for i in range(len(self._ret.eigenvalues)):
-                eigvec = self._ret.eigenstates[i]
-                eigval = self._ret.eigenvalues[i]
-                if self._ret.aux_operators_evaluated is not None:
-                    aux_op = self._ret.aux_operators_evaluated[i]
+            for i in range(len(result.eigenvalues)):
+                eigvec = result.eigenstates[i]
+                eigval = result.eigenvalues[i]
+                if result.aux_operators_evaluated is not None:
+                    aux_op = result.aux_operators_evaluated[i]
                 else:
                     aux_op = None
                 if self._filter_criterion(eigvec, eigval, aux_op):
                     cnt += 1
                     eigvecs += [eigvec]
                     eigvals += [eigval]
-                    if self._ret.aux_operators_evaluated is not None:
+                    if result.aux_operators_evaluated is not None:
                         aux_ops += [aux_op]
                 if cnt == k_orig:
                     break
 
-            self._ret.eigenstates = np.array(eigvecs)
-            self._ret.eigenvalues = np.array(eigvals)
+            result.eigenstates = np.array(eigvecs)
+            result.eigenvalues = np.array(eigvals)
             # conversion to np.array breaks in case of aux_ops
-            self._ret.aux_operators_evaluated = aux_ops
+            result.aux_operators_evaluated = aux_ops
 
             self._k = k_orig
 
-        # evaluate ground state after filtering (in case a filter is set)
-        self._get_ground_state_energy(operator)
-        if self._ret.eigenstates is not None:
-            self._ret.eigenstates = [Statevector(vec) for vec in self._ret.eigenstates]
+        if result.eigenstates is not None:
```

Comment:
It seems that this is never None, so would it make sense to immediately cast to a `Statevector` above?